### PR TITLE
mvapich: init at 2.3.6

### PR DESCRIPTION
**Pull request by markuskowa**
###### Motivation for this change
Another MPI implementation optimized for Infiniband networks.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/mvapich/default.nix [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    sed -i 's:CC="gcc":CC=${stdenv.cc}/bin/gcc:' $out/bin/mpicc
+    sed -i 's:CXX="g++":CXX=${stdenv.cc}/bin/g++:' $out/bin/mpicxx
+    sed -i 's:FC="gfortran":FC=${gfortran}/bin/gfortran:' $out/bin/mpifort
+  '';
+
+  enableParallelBuilding = true;
+
+  meta = with lib; {
+    description = "MPI-3.1 implementation optimized for Infiband transport";
+    homepage = "http://mvapich.cse.ohio-state.edu";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/mvapich/default.nix:68]:
```suggestion
    homepage = "https://mvapich.cse.ohio-state.edu/";
```

File: pkgs/development/libraries/mvapich/default.nix [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    # /tmp/nix-build... ends up in the RPATH, fix it manually
+    for entry in $out/bin/mpichversion $out/bin/mpivars; do
+      echo "fix rpath: $entry"
+      patchelf --set-rpath "$out/lib" $entry
+    done
+
+    # Ensure the default compilers are the ones mvapich was built with
+    sed -i 's:CC="gcc":CC=${stdenv.cc}/bin/gcc:' $out/bin/mpicc
+    sed -i 's:CXX="g++":CXX=${stdenv.cc}/bin/g++:' $out/bin/mpicxx
+    sed -i 's:FC="gfortran":FC=${gfortran}/bin/gfortran:' $out/bin/mpifort
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/mvapich/default.nix:61]: Please use substituteInPlace.

File: pkgs/development/libraries/mvapich/default.nix [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+  meta = with lib; {
+    description = "MPI-3.1 implementation optimized for Infiband transport";
+    homepage = "http://mvapich.cse.ohio-state.edu";
+    license = licenses.bsd3;
+    maintainers = [ maintainers.markuskowa ];
+    platforms = platforms.linux;
+  };
+}
+
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/mvapich/default.nix:74]: One final new line to much.

File: pkgs/development/libraries/mvapich/default.nix [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+stdenv.mkDerivation rec {
+  pname = "mvapich";
+  version = "2.3.6";
+
+  src = fetchurl {
+    url = "http://mvapich.cse.ohio-state.edu/download/mvapich/mv2/mvapich2-${version}.tar.gz";
+    sha256 = "0jd28vy9ivl3rcpkxmhw73b6krzm0pd9jps8asw92wa00lm2z9mk";
+  };
+
+  nativeBuildInputs = [ pkgconfig bison makeWrapper ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/mvapich/default.nix:24]:
```suggestion
  nativeBuildInputs = [ pkg-config bison makeWrapper ];
```

File: pkgs/development/libraries/mvapich/default.nix [hunk 5]
```diff
@@ -0,0 +1,74 @@
+{ lib, stdenv, fetchurl, pkgconfig, bison, numactl, libxml2
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/mvapich/default.nix:1]:
```suggestion
{ lib, stdenv, fetchurl, pkg-config, bison, numactl, libxml2
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by markuskowa** [comment]
All cleared.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 125267` at 95bcd74d run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>mvapich</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md">stale-substitute</a>

Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'CC="gfortran"' doesn't match anything in file '/nix/store/2ifxld2sh6d6y98c06fk6mb0xhmvapv4-mvapich-2.3.6/bin/mpifort'
Near pkgs/development/libraries/mvapich/default.nix:67:0:
```
   |
67 |     description = "MPI-3.1 implementation optimized for Infiband transport";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md">stale-substitute</a>

Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'CC="g++"' doesn't match anything in file '/nix/store/2ifxld2sh6d6y98c06fk6mb0xhmvapv4-mvapich-2.3.6/bin/mpicxx'
Near pkgs/development/libraries/mvapich/default.nix:67:0:
```
   |
67 |     description = "MPI-3.1 implementation optimized for Infiband transport";
   | ^
```
</li>
  </ul>
</details>

**Pull request merged**